service/biz/enrollz_biz.go

```diff
@@ -463,19 +465,247 @@ func RotateAIKCert(ctx context.Context, req *RotateAIKCertReq) error {
 		log.ErrorContext(ctx, err)
 		return err
 	}
-	// TODO: Generate Issuer Key Pair
-	// TODO: Send issuer_public_key to the device.
-	// TODO: Handle application_identity_request
-	// TODO: Decrypt application_identity_request
-	// TODO: Parse application_identity_request
+	// Generate Issuer Key Pair
+	issuerPrivateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		err = fmt.Errorf("failed to generate issuer key pair: %w", err)
+		log.ErrorContext(ctx, err)
+		return err
+	}
+	issuerPublicKey := &issuerPrivateKey.PublicKey
+
+	// Send issuer_public_key to the device.
+	publicKeyBytes, err := x509.MarshalPKIXPublicKey(issuerPublicKey)
+	if err != nil {
+		err = fmt.Errorf("failed to marshal issuer public key: %w", err)
+		log.ErrorContext(ctx, err)
+		return err
+	}
+	rotateAIKCertReq := &epb.RotateAIKCertRequest{
+		Value: &epb.RotateAIKCertRequest_IssuerPublicKey{
+			IssuerPublicKey: publicKeyBytes,
+		},
+		ControlCardSelection: req.ControlCardSelection,
+	}
+
+	stream, err := req.Deps.RotateAIKCert(ctx)
+	if err != nil {
+		err = fmt.Errorf("failed to initiate RotateAIKCert stream: %w", err)
+		log.ErrorContext(ctx, err)
+		return err
+	}
+	if err := stream.Send(rotateAIKCertReq); err != nil {
+		err = fmt.Errorf("failed to send issuer public key to the device: %w", err)
+		log.ErrorContext(ctx, err)
+		return err
+	}
+
+	// Handle application_identity_request
+	resp, err := stream.Recv()
+	if err != nil {
+		err = fmt.Errorf("failed to receive application_identity_request from the device: %w", err)
+		log.ErrorContext(ctx, err)
+		return err
+	}
+	applicationIdentityRequestBytes := resp.GetApplicationIdentityRequest()
+	if len(applicationIdentityRequestBytes) == 0 {
+		err = fmt.Errorf("application_identity_request is empty")
+		log.ErrorContext(ctx, err)
+		return err
+	}
+
+	// Decrypt application_identity_request
+	decryptedApplicationIdentityRequestBytes, err := DecryptWithPrivateKey(ctx, issuerPrivateKey, applicationIdentityRequestBytes)
+	if err != nil {
+		err = fmt.Errorf("failed to decrypt application_identity_request: %w", err)
+		log.ErrorContext(ctx, err)
+		return err
+	}
+
+	// Parse application_identity_request
+	applicationIdentityRequest, err := ParseIdentityRequest(decryptedApplicationIdentityRequestBytes)
+	if err != nil {
+		err = fmt.Errorf("failed to parse identity proof: %w", err)
```

Comment:
to keep with the verbiage above, maybe s/identity proof/application_identity_request/ or via versa?